### PR TITLE
[processor/remotetap] Fix memory leak on shutdown

### PR DESCRIPTION
**Pull request by crobert-1**
**Description:** <Describe what has changed.>

The remotetap processor holds a map of channels for writing telemetry data to WebSockets when a client connects. The write process works by blocking on a given channel waiting for telemetry data coming through the processor's pipeline, then writing as soon as it comes in. This however can result in a leaked goroutine that blocks forever if data is not received, which is always the case when shutting down.

This fixes the bug by closing and deleting all channels once all client connections have been shutdown gracefully. This fixes the blocking behavior during shutdown.

This change also enables `goleak` to help ensure no goroutines are being leaked.

**Link to tracking Issue:** <Issue number if applicable>
#30438

**Testing:** <Describe what testing was performed and which tests were added.>
All existing tests are passing, as well as added `goleak` check.

### CODE REVIEW

File: .chloggen/goleak_remotetap.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: remotetapprocessor
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Fix memory leak on shutdown
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: []
```
Comment (crobert-1) [inline .chloggen/goleak_remotetap.yaml:13]:
```suggestion
issues: [32571]
```

**Review by atoulme — APPROVED** [review]

**Review by songy23 — APPROVED** [review]

File: processor/remotetapprocessor/metadata.yaml [hunk 2]
```diff
@@ -13,4 +13,4 @@ status:
 tests:
   config:
   goleak:
-    skip: true
+    skip: false
```
Comment (codeboten) [inline processor/remotetapprocessor/metadata.yaml:16]: the goleak section can be removed completely here

**Review by codeboten — CHANGES_REQUESTED** [review]

File: processor/remotetapprocessor/metadata.yaml (hunk 2 above)
Comment (crobert-1) [inline processor/remotetapprocessor/metadata.yaml:16]:
```suggestion
```

File: processor/remotetapprocessor/metadata.yaml [hunk 3]
```diff
@@ -13,4 +13,3 @@ status:
 tests:
   config:
   goleak:
```
Comment (codeboten) [inline processor/remotetapprocessor/metadata.yaml:15]: This can be removed as well :)

Comment (crobert-1) [reply processor/remotetapprocessor/metadata.yaml]: 🤦 Done

**Review by codeboten — APPROVED** [review]

**Pull request merged**